lib/postgrex/extension.ex

```diff
@@ -67,14 +90,31 @@ defmodule Postgrex.Extension do
   @callback format(opts) :: :binary | :text
 
   @doc """
-  Should encode an Elixir value to a binary in the specified Postgres protocol
-  format.
+  Returns a quoted list of clauses that encode an Elixir value to iodata with
+  a signed 32 bit big endian integer byte length header.
+
+      def encode(_) do
+        quote do
+          integer ->
+            <<8 :: signed-32, integer :: signed-64>>
+        end
+      end
+
   """
-  @callback encode(TypeInfo.t, term, Types.types, opts) :: iodata
+  @callback encode(opts) :: Macro.expr
 
   @doc """
-  Should decode a binary in the specified Postgres protocol format to an Elixir
-  value.
+  Returns a quoted list of clauses that decode a binary to an Elixir value. The
```

Comment:
The first paragraph must always be a short summary. Just start a paragraph after the first sentence. :)